GEE_pull_functions.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python2
+# -*- coding: utf-8 -*-
+"""
+Google Earth Engine Reflectance Pull Functions
+
+@author: simontopp
+"""
+import ee
+import time
+import os
+
+## Recreate the fmask based on Collection 2 Pixel QA band
+def AddFmask(image):
+    qa = image.select('pixel_qa')
+    water = qa.bitwiseAnd(1 << 7)
+    cloud = qa.bitwiseAnd(1 << 1).Or(qa.bitwiseAnd(1 << 2)).Or(qa.bitwiseAnd(1 << 3))
+    snow = qa.bitwiseAnd(1 << 5)
+    cloudshadow = qa.bitwiseAnd(1 << 4)
+
+    fmask = (water.gt(0).rename('fmask')
+             .where(snow.gt(0), ee.Image(3))
+             .where(cloudshadow.gt(0), ee.Image(2))
+             .where(cloud.gt(0), ee.Image(4))
+             .updateMask(qa.gte(0)))
+    # mask the fmask so that it has the same footprint as the quality (BQA) band
+    return image.addBands(fmask)
+
+
+## Buffer the lake sites either 120 minutes or the distance to shore, whichever
+## is smaller
+def dpBuff(i):
+    dist = i.get('distance')
+    buffdist = ee.Number(dist).min(120)
+    return i.buffer(buffdist)
+
+
+## Remove geometries
+def removeGeo(i):
+    return i.setGeometry(None)
+
+
+## Set up the reflectance pull
+def RefPull(image):
```

Comment:
should/can this function accept `era5` as an argument, given that it's created in the other file and used in here? Same with `lakes`?

Reply:
Another EE quirk, when you use the EE map function you can only pass one argument to it. In other words, you can only map over either an ImageCollection or FeatureCollections.  Thats why lakes and era5 are defined elsewhere.